nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/JCAMPData.java

```diff
@@ -17,987 +17,569 @@
+        return block.optional($TD, 1)
+                .map(r -> r.getInt() / 2)
+                .orElse(1);
     }
 
     @Override
-    public Double getParDouble(String parname) {
-        if ((parMap == null) || (parMap.get(parname) == null)) {
-            return null;
-//            throw new NullPointerException();
-        } else {
-            return Double.parseDouble(parMap.get(parname));
-        }
+    public int getNPoints() {
+        // XXX original JCAMPData always returned 0 (np / 2, with np never initialized). Isn't it a bug?
+
+        // XXX always /2, or only if complex data?
+        // XXX should we try to get it from page content instead?
+        return block.optional($TD, 0)
+                .map(r -> r.getInt() / 2)
+                .orElse(0);
     }
 
     @Override
-    public Integer getParInt(String parname) {
-        if ((parMap == null) || (parMap.get(parname) == null)) {
-            return null;
-        } else {
-            return Integer.parseInt(parMap.get(parname));
-        }
+    public int getNDim() {
+        return block.getOrDefault(NUMDIM, "1").getInt();
     }
 
     @Override
-    public int getNVectors() {  // number of vectors
-        return nvectors;
+    public int getSize(int dim) {
+        // XXX should we try to get it from page content instead?
+        // Shouldn't the "/2" be only for complex data?
+        return block.optional($TD, dim)
+                .map(r -> r.getInt() / 2)
+                .orElse(0);
     }
 
     @Override
-    public int getNPoints() {  // points per vector
-        return np / 2;
+    public void setSize(int dim, int size) {
+        //TODO implement me?
+        // Don't see how we would need this as we are accessing the size directly
+        System.out.println("Called unimplemented method: setSize: " + dim + ", " + size);
     }
 
     @Override
-    public int getNDim() {
-        return dim;
+    public String getSolvent() {
+        return block.optional(_SOLVENT_NAME, $SOLVENT)
+                .map(JCampRecord::getString)
+                .orElse("");
     }
 
     @Override
-    public boolean isComplex(int iDim) {
-        return complexDim[iDim];
+    public double getTempK() {
+        return block.optional(TEMPERATURE, $TE)
+                .map(r -> r.getDouble() > 150 ? r.getDouble() : 273.15 + r.getDouble())
+                .orElse(298.0); //XXX default value was already a question in original JCAMP data
     }
 
-    // used only for indirect dimensions; direct dimension done when FID read 
     @Override
-    public boolean getNegatePairs(int iDim) {
-        return (fttype[iDim].equals("negate"));
+    public String getSequence() {
+        return block.optional(_PULSE_SEQUENCE, $PULPROG)
+                .map(JCampRecord::getString).orElse("");
     }
 
     @Override
-    public boolean getNegateImag(int iDim) {
-        return iDim > 0;
+    public double getSF(int dim) {
```

Comment:
Should SF be BASE_FREQ (without offset) or OBSERVED_FREQ? I chose the latter when possible.